tools/create_service_bundle.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python3
+#
+# TODO: usage, description.
+
+from datetime import date, datetime
+from typing import List
+import json
+import logging
+import os
+import re
+import retrying
+import sys
+
+from sdk_utils import groupby
+import sdk_cmd
+import sdk_diag
+import sdk_utils
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+DEFAULT_RETRY_WAIT = 1000
+DEFAULT_RETRY_MAX_ATTEMPTS = 5
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def _get_task_page(
+    offset: int, limit: int, framework_id: str = None, task_id: str = None
+) -> dict:
+    path = "/mesos/tasks?offset={}&limit={}".format(offset, limit)
+
+    if framework_id:
+        path += "&framework_id={}".format(framework_id)
+
+    if task_id:
+        path += "&task_id={}".format(task_id)
+
+    return sdk_cmd.cluster_request("GET", path).json()
+
+
+def _get_tasks(
+    tasks: dict, offset: int, limit: int, framework_id: str = None, task_id: str = None
+) -> dict:
+    page = _get_task_page(offset, limit, framework_id=framework_id, task_id=task_id)
+    page_tasks = page.get("tasks", [])
+    tasks_tasks = tasks.get("tasks", [])
+    merged_tasks = tasks_tasks + page_tasks
+    tasks["tasks"] = merged_tasks
+
+    if len(page_tasks) < limit:
+        return tasks
+    else:
+        return _get_tasks(
+            tasks, offset + limit, limit, framework_id=framework_id, task_id=task_id
+        )
+
+
+def get_tasks(framework_id: str = None, task_id: str = None) -> dict:
+    offset = 0
+    limit = 100
+    tasks = {}
+
+    return _get_tasks(tasks, offset, limit, framework_id=framework_id, task_id=task_id)
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def marathon_app_show(marathon_app_name: str) -> dict:
+    return sdk_cmd.get_json_output(
+        "marathon app show {}".format(marathon_app_name), print_output=False
+    )
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def get_scheduler_tasks(app: dict) -> dict:
+    return list(
+        map(lambda task: get_tasks(task_id=task["id"])["tasks"][0], app["tasks"])
+    )
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def get_framework_id(package_name: str, service_name: str) -> str:
+    framework_ids = sdk_cmd.svc_cli(
+        package_name,
+        service_name,
+        "debug state framework_id",
+        json=True,
+        print_output=False,
+    )
+
+    assert len(framework_ids) == 1, "More than 1 Framework ID returned: {}".format(
+        framework_ids
+    )
+
+    return framework_ids[0]
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def debug_agent_files(agent_id: str) -> List[str]:
```

Comment:
Are these already part of the `dcos-diagnostics` bundle?

Reply:
I'm not sure what you mean. This function is used to list the files on an agent so that the script can match the executor path. The files are not downloaded though.